zio-cache/shared/src/main/scala/zio/cache/ScopedCache.scala

```diff
@@ -99,41 +99,44 @@ object ScopedCache {
   def makeWith[Key, Environment, Error, Value](
     capacity: Int,
     scopedLookup: ScopedLookup[Key, Environment, Error, Value]
-  )(timeToLive: Exit[Error, Value] => Duration): ZIO[Environment with Scope, Nothing, ScopedCache[Key, Error, Value]] =
+  )(timeToLive: Exit[Error, Value] => Duration): URIO[Environment with Scope, ScopedCache[Key, Error, Value]] =
     makeWith(capacity, scopedLookup, Clock.systemUTC())(timeToLive)
 
   //util for test because it allow to inject a mocked Clock
   private[cache] def makeWith[Key, Environment, Error, Value](
     capacity: Int,
     scopedLookup: ScopedLookup[Key, Environment, Error, Value],
     clock: Clock
-  )(timeToLive: Exit[Error, Value] => Duration): ZIO[Environment with Scope, Nothing, ScopedCache[Key, Error, Value]] =
+  )(timeToLive: Exit[Error, Value] => Duration): URIO[Environment with Scope, ScopedCache[Key, Error, Value]] =
     ZIO
-      .acquireRelease(buildWith(capacity, scopedLookup, clock)(timeToLive)) { case (scopedCache, daemonFiber) =>
-        scopedCache.invalidateAll *> daemonFiber.interrupt.unit
+      .acquireRelease(buildWith(capacity, scopedLookup, clock)(timeToLive))(_.invalidateAll)
+      .flatMap { scopedCache =>
+        runFreeExpiredResourcesLoopInBackground(scopedCache)
+          .as(scopedCache)
       }
-      .map { case (scopedCache, _) => scopedCache }
+
+  private def runFreeExpiredResourcesLoopInBackground[Key, Environment, Error, Value](
+    scopedCache: ScopedCacheImplementation[Key, Error, Value, Environment]
+  ): URIO[Scope, Fiber.Runtime[Nothing, Long]] = {
+    val cleaningInterval = 1.second
+    (scopedCache.freeExpired *> ZIO.sleep(cleaningInterval))
+      .repeat(Schedule.forever)
+      .interruptible
```

Comment:
Good use of `interruptible` here.

Reply:
Honestly I am never sure when or when not I need to use `interruptible`. I just saw that without it my unit tests didn't pass. Are fiber `uninterruptible by default`?

Reply:
Fibers are interruptible by default but they inherit the interruptibility of their parent so if you fork a fiber in an uninterruptible region and then try to interrupt it you will suspend indefinitely.